### PR TITLE
adding windows cursor strings

### PR DESCRIPTION
**Pull request by jasonwilliams**
Hey guys, ad @vvuk  i've started adding in the windows string for the cursor.
Could you advise on what else needs doing here or how i can test this is working?

I can't seem to find some of the windows cursor actions, such as grabbing.
been looking here:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms648391(v=vs.85).aspx

Thanks

### CODE REVIEW

**Comment by fkaa** [comment]
You can test it with `cargo run --example cursor`. You don't seem to be actually using the `LoadCursor` function you linked though (The OSX stuff is completely separate from the win32 backend)

**Comment by vvuk** [comment]
Good start! Looks like you copied the osx version of this finction to start?

Instead of strings, you want to use the enums from the winapi module
directly, e.g. winapi::IDC_ARROW. Then, you seem to have the OSX code
copied (which does objc message sending)... instead, you need to call the
win32 LoadCursor function like is done later on in the file. Also, the
comments still reference OSX names -- should either nuke them or update
them for windows.

**Comment by Jayflux** [comment]
hey guys
thanks for the feedback, ive made some updates. please see the changed file.
Whats left to do on this?

File: src/api/win32/mod.rs [hunk 1]
```diff
@@ -253,8 +253,35 @@ impl Window {
...
+            MouseCursor::NResize => winapi::IDC_SIZENS,
+            MouseCursor::WResize => winapi::IDC_SIZEWE,
+            MouseCursor::SResize => winapi::IDC_SIZENS,
+            MouseCursor::EwResize | MouseCursor::ColResize => winapi::IDC_SIZEWE,
+            MouseCursor::NsResize | MouseCursor::RowResize => winapi::IDC_SIZENS,
+
+            MouseCursor::Grabbing | MouseCursor::Grab | MouseCursor::Copy |
+            MouseCursor::Alias    | MouseCursor::ContextMenu |
+            MouseCursor::NeResize | MouseCursor::NwResize |
+            MouseCursor::SeResize | MouseCursor::SwResize |
```
Comment (retep998) [inline src/api/win32/mod.rs]: Use `IDC_SIZENESW` and `IDC_SIZENWSE` for these.

File: src/api/win32/mod.rs [hunk 2]
```diff
@@ -253,8 +253,35 @@ impl Window {
...
+            MouseCursor::NwseResize | MouseCursor::NeswResize |
```
Comment (retep998) [inline src/api/win32/mod.rs]: Presumably these should be `IDC_SIZEALL`

File: src/api/win32/mod.rs [hunk 3]
```diff
@@ -253,8 +253,35 @@ impl Window {
...
+
+            MouseCursor::Cell | MouseCursor::NoneCursor |
+            MouseCursor::Wait | MouseCursor::Progress | MouseCursor::Help |
```
Comment (retep998) [inline src/api/win32/mod.rs]: Wait and Progress should likely be `IDC_WAIT`

File: src/api/win32/mod.rs (hunk 1 above)
Comment (Jayflux) [inline src/api/win32/mod.rs]: @retep998 could you be more specific?
You've quoted the whole code block here

File: src/api/win32/mod.rs (hunk 2 above)
Comment (Jayflux) [inline src/api/win32/mod.rs]: @retep998 could you be more specific?
You've quoted the whole code block here

Comment (retep998) [inline src/api/win32/mod.rs]: NwseResize and NeswResize

File: src/api/win32/mod.rs (hunk 1 above)
Comment (retep998) [inline src/api/win32/mod.rs]: NeResize and SwResize -> `IDC_SIZENESW`
NwResize and SeResize -> `IDC_SIZENWSE`

**Comment by Jayflux** [comment]
Updated....
Let me know what needs updating @vvuk @retep998

File: src/api/win32/mod.rs [hunk 4]
```diff
@@ -253,8 +253,35 @@ impl Window {
...
+            MouseCursor::Wait | MouseCursor::Progress => winapi::IDC_WAIT,
+            MouseCursor::Help => winapi::IDC_HELP,
+
+            MouseCursor::Grabbing | MouseCursor::Grab | MouseCursor::Copy |
+            MouseCursor::Alias    | MouseCursor::ContextMenu |
+            MouseCursor::Cell | MouseCursor::NoneCursor |
+            MouseCursor::Move | MouseCursor::AllScroll | MouseCursor::ZoomIn |
+            MouseCursor::ZoomOut => winapi::IDC_ARROW,
+        };
+            unsafe {
```
Comment (tomaka) [inline src/api/win32/mod.rs]: Wrong identation here

**Comment by Jayflux** [comment]
Anyone know why this still doesn't work?

**Comment by Jayflux** [comment]
Could have something to do with this:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms648393(v=vs.85).aspx

> Calling SetCursor on a non-GUI thread does not appear to work. The simple fix is surrounding the call with AttachThreadInputs of the current thread to the GetForegroundWindow thread. I picked this up from: http://www.codeproject.com/KB/cpp/WaitCursor.aspx

> example:

> HWND hWnd = GetForegroundWindow();
DWORD foregroundThreadID = GetWindowThreadProcessId(hWnd, 0);
DWORD currentThreadID = GetCurrentThreadId();

> AttachThreadInput(foregroundThreadID, currentThreadID, TRUE);
SetCursor(hCursor);
AttachThreadInput(foregroundThreadID, currentThreadID, FALSE);

**Comment by Jayflux** [comment]
@vvuk @fkaa  should we create a new ticket to address this issue?

**Comment by francesca64** [comment]
This has long since been implemented in winit.

**Pull request closed without merging**